src/main/java/core/basesyntax/service/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service;
+
+import core.basesyntax.Colors;
+import core.basesyntax.Figure;
+import core.basesyntax.Figures;
+import core.basesyntax.entity.Circle;
+import core.basesyntax.entity.IsoscelesTrapezoid;
+import core.basesyntax.entity.Rectangle;
+import core.basesyntax.entity.RightTriangle;
+import core.basesyntax.entity.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        Figure figure = null;
+        int randomFigureNumber = new Random().nextInt(Figures.values().length);
+
+        switch (Figures.values()[randomFigureNumber]) {
+            case SQUARE:
+                figure = new Square();
```

Comment:
This is where the logic for creating random figures should be, not in the constructor. Also, move the figure creation to private methods and use here